app/src/main/java/at/roteskreuz/stopcorona/model/repositories/QuarantineRepository.kt

```diff
@@ -366,6 +395,44 @@ class QuarantineRepositoryImpl(
             quarantineStateObservable.blockingFirst()
         }
     }
+
+    override fun observeIfUploadOfMissingExposureKeysIsNeeded(): Observable<Optional<UploadMissingExposureKeys>> {
+        return Observables.combineLatest(
+            observeDateOfFirstMedicalConfirmation(),
+            preferences.observeNullableZonedDateTime(PREF_DATE_OF_LAST_SELF_DIAGNOSE),
+            preferences.observeBoolean(PREF_MISSING_KEYS_UPLOADED, false)
+        ).map { (dateOfFirstMedicalConfirmation, dateOfLastSelfDiagnose, areMissingKeysUploaded) ->
```

Comment:
Add `.debounce(50, TimeUnit.MILLISECONDS) // some of the shared prefs can be changed together`

Reply:
updated